pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1152,171 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func runningOnKVM() bool {
+	_, enabledHVs := hypervisor.GetAvailableHypervisors()
+	for _, enabledHV := range enabledHVs {
+		if enabledHV == hypervisor.KVMHypervisorName {
+			return true
+		}
+	}
+	return false
+}
+
+func getHostCpusNum() (int, error) {
+	if !runningOnKVM() {
+		log.Warnf("Running not on KVM/QEMU system, skip getting CPUs info")
+		return 0, nil
+	}
+	// The `nproc` util relies on the /proc/sys file system, that is not properly accessible from a container.
+	// So, have to run the command with access to the host FS.
+	out, err := exec.Command("chroot", "/hostfs", "nproc", "--all").Output()
+	if err != nil {
+		return -1, err
+	}
+	// remove "\n"
+	out = out[:2]
+	maxHostCpus, _ := strconv.Atoi(string(out))
```

Comment:
I would recommend using `strings.TrimSpace` instead of slicing, just so EVE does not panic if it receives some unexpected short string.
Also let's not hide any potential error from `strconv.Atoi`, but return it.
I realize none should be reachable, this is just in case...

Reply:
done